### ISSUE TITLE
Constants declared by using object/array binding pattern syntax fail the build

### ISSUE THREAD

**Issue by thefat32** [timocov/dts-bundle-generator#315]
**Bug report**

When importing "single-spa" package (in this case "single-spa-react" is importing "single-spa") dts-bundle-generator fails with error:

![image](https://github.com/timocov/dts-bundle-generator/assets/10998637/6442b7ff-6c1d-4bc6-ba26-e9b005d06bbe)

**Input code**

```ts
import React from "react";
import ReactDOMClient from "react-dom/client";
import singleSpaReact from "single-spa-react";

const lifecycles = singleSpaReact({
  React,
  ReactDOMClient,
  rootComponent: () => "HELLO WORLD!",
  errorBoundary(err, info, props) {
    console.error("Error Boundary", err, info, props);
    return React.createElement('div', { className: 'root' }, 'ERROR BOUNDARY');
  },
});

export const { bootstrap, mount, unmount } = lifecycles;

```

**Expected output**

Not to fail

**Actual output**

Fails with Error: Cannot find symbol for node...

**Additional context**

Resproduction repo:

https://github.com/thefat32/dts-bundle-generator-single-spa-bug

The error is beacuse of this export syntax in `node_modules/single-spa/typings/single-spa.d.ts`

```ts
  export const {
    NOT_LOADED = "NOT_LOADED",
    LOADING_SOURCE_CODE = "LOADING_SOURCE_CODE",
    NOT_BOOTSTRAPPED = "NOT_BOOTSTRAPPED",
    BOOTSTRAPPING = "BOOTSTRAPPING",
    NOT_MOUNTED = "NOT_MOUNTED",
    MOUNTING = "MOUNTING",
    MOUNTED = "MOUNTED",
    UPDATING = "UPDATING",
    UNMOUNTING = "UNMOUNTING",
    UNLOADING = "UNLOADING",
    SKIP_BECAUSE_BROKEN = "SKIP_BECAUSE_BROKEN",
    LOAD_ERROR = "LOAD_ERROR",
  };
```

In `single-spa`'s [repo](https://github.com/single-spa/single-spa/blob/5e45ae8fa2308824339d85e4a9409f6199eba099/typings/single-spa.d.ts#L178)

It's a strange syntax I have never seen it, but it seems to be interpreted correctly by typescript. As an axample this typescript code compiles without erros in typescript: 

```ts
import { NOT_LOADED } from "single-spa";

console.log(NOT_LOADED);

// prints "NOT_LOADED"
```

**Comment by timocov** [comment]
> The error is beacuse of this export syntax in node_modules/single-spa/typings/single-spa.d.ts

Omg what the hell is this... I don't know what the compiler even allows to write it this way, this looks like a bug and shouldn't be allowed 😄

Anyway, I'll take a look what we can do to avoid an error from happening (but no promises, this syntax is very exotic and I need to play with it a bit to see what else could be there apart from just const values).

**Comment by thefat32** [comment]
At first I thought it was a bug in the `single-spa.d.ts` file... I don't know why typescript compiler is allowing this syntax. Couldn't find any info about it.

I've created an issue in `single-spa`'s repo to ask them to stop using this exotic/bugged syntax.

single-spa/single-spa#1215

**Comment by timocov** [comment]
I also [asked in the typescript discord chat](https://discord.com/channels/508357248330760243/1229206938018119710/1229206938018119710) about this syntax, will see what people would say...

**Comment by timocov** [comment]
There is a non-zero chance that this cannot be supported by the tool (without writing a ton of code). Currently I'm leaning towards putting a warning if the tool encountered such syntax. The build won't fail with the error above, but the resulted dts might or might not be correct in this case (which will be warned by the tool ofc). The only possible solution would be to change the source code.

From my testing I couldn't find a way how the compiler would generate such code (any object/array binding is converted into "variable declaration per a variable" syntax which is much easier to parse and I believe it is handled already).

**Comment by thefat32** [comment]
I think single-spa.d.ts file is being updated manually by the single-spa team. That's why it has that weird syntax.

**Comment by timocov** [comment]
> I think detecting this kind of node and expanding it to multiple nodes should fix the problem

I wish it was just so easy to do... Btw apart from splitting a variable declaration list into multiple variable declarations this is exactly what the compiler does when you write similar code but in `.ts` file.

The issue is, the tool doesn't transform source files nor even their compiled versions. In order to perform any transformation the only option I'm aware of requires to run a compilation with transformation, but the issue is - the compiler doesn't transform `.d.ts` files because they are treated as "output" already thus not being transformed. Furthermore, such declaration can be in `node_modules` which is ignored from output generation regardless.

I might have a solution tho - it looks like the only option is to "print" source file and parse it again, which might and most likely will have performance implications (we can try to reduce it, but it still would have some cost to check when we need to do it).

**Comment by timocov** [comment]
The fix has been published in [v9.5.0](https://github.com/timocov/dts-bundle-generator/releases/tag/v9.5.0).

**Comment by thefat32** [comment]
@timocov impressive work, thank you very much
